tools/test_containers_compose

```diff
@@ -20,7 +20,7 @@ wait_until() {
 
 setup_containers() {
   for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done
-  sudo MOJO_CLIENT_DEBUG=1 docker-compose up -d
+  sudo MOJO_CLIENT_DEBUG=1 docker-compose up -d || docker-compose logs && exit 1
```

Comment:
This is wrong as well. We don't want to exit with code 1 when `docker-compose up` succeeded. (Mind the operator precedence. Likely it is best to use
```
if ! sudo …; then
    exit_code=$?
    echo "docker-compose exited with non-zero code $exit_code, showing logs:"
    docker-compose logs
    exit "$exit_code"
fi
```
here.)

Reply:
Thanks fixed.

Reply:
and please provide tests for all introduced branches. If not possible automatically then please describe the manual tests done for verification in the git commit message